Sources/Document/CRDT/CRDTElement.swift

```diff
@@ -105,64 +114,30 @@ class CRDTElement {
         return false
     }
 
-    /**
-     * `isRemoved` check if this element was removed.
-     */
-    func isRemoved() -> Bool {
-        return self.removedAt != nil
-    }
-
-    func toJSON() -> String {
-        fatalError("Must be implemented.")
-    }
-
-    func toSortedJSON() -> String {
-        fatalError("Must be implemented.")
-    }
-
-    func deepcopy() -> CRDTElement {
-        fatalError("Must be implemented.")
-    }
-}
-
-extension CRDTElement: Equatable {
-    static func == (lhs: CRDTElement, rhs: CRDTElement) -> Bool {
-        return lhs.createdAt == rhs.createdAt && lhs.movedAt == rhs.movedAt && lhs.removedAt == rhs.removedAt
+    func equal(_ target: CRDTElement) -> Bool {
+        return self.createdAt == target.createdAt && self.movedAt == target.movedAt && self.removedAt == target.removedAt
     }
 }
 
 /**
  *
  * `CRDTContainer` represents CRDTArray or CRDtObject.
- * @internal
  */
-class CRDTContainer: CRDTElement {
-    func subPath(createdAt: TimeTicket) throws -> String {
-        fatalError("Must be implemented.")
-    }
+protocol CRDTContainer: CRDTElement {
+    func subPath(createdAt: TimeTicket) throws -> String
 
-    func purge(element: CRDTElement) throws {
-        fatalError("Must be implemented.")
-    }
+    func purge(element: CRDTElement) throws
```

Comment:
@hackerwins How about replacing purge with delete?

Reply:
I'd love to discuss this with @7hong13.

Reply:
References) CRDTArray/purge -> RGATreeList/purge -> RGATreeList/release -> RGATreeListNode/release & SplayTree/delete

Reply:
unifying function name into one sounds good to me. I think `purge` has more similar meaning to `release`.
(`delete` in RGATreeList is not unlinking the node, but just marking it as removed.)

Reply:
I think delete of RGATreeList should be changed to remove. (Swift changed)

Reply:
that seems good to me. I will change function names in kotlin as below:
purge -> delete
delete -> remove